src/btree/bt_cursor.c

```diff
@@ -778,6 +832,53 @@ __wt_btcur_search(WT_CURSOR_BTREE *cbt)
     return (ret);
 }
 
+/*
+ * __btcur_search_near_bounds_reposition --
+ *     Search near with a bounded cursor should reposition the cursor at the nearest bound.
+ */
+static int
+__btcur_search_near_bounds_reposition(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt)
+{
+    WT_BTREE *btree;
+    WT_CURSOR *cursor;
+    bool upper_out_of_bounds, lower_out_of_bounds;
+
+    btree = CUR2BT(cbt);
+    cursor = &cbt->iface;
+    lower_out_of_bounds = upper_out_of_bounds;
+
+    WT_ASSERT(session, WT_CURSOR_HAS_BOUNDS(cursor));
+
+    if (F_ISSET(cursor, WT_CURSTD_BOUND_LOWER)) {
+        if (btree->type == BTREE_ROW)
+            /*
+             * We need to assume that the bounds are inclusive here, suppose a caller calls with the
+             * search key set to the lower bound but also specifies that the lower bound isn't
+             * inclusive. We cannot know which key to set the lower bound to so we set it to the
+             * lower bound. The same is true for the upper bound check.
+             */
+            WT_RET(__wt_row_compare_bounds(
```

Comment:
I would prefer to set the cursor inclusive flags and restore the cursor state flag, instead of adding a new argument.

Reply:
Personally I prefer not to do that as it means that the caller needs to make sure that they correctly restore the right flag state.